src/groups.py

```diff
@@ -1,15 +1,27 @@
 
+# all dependancy on nodeattr is to be located in this file
 from plumbum import local
-
-__nodeattr = local['nodeattr']['-f', '/var/lib/adminware/genders']
+from tempfile import NamedTemporaryFile
 
 def list():
-    groups = __nodeattr('-l').split("\n")
+    groups = __nodeattr()('-l').split("\n")
     groups.remove('')
     return groups
 
 def nodes_in(group_name):
-    nodes = __nodeattr('-n', group_name).split("\n")
+    nodes = __nodeattr()('-n', group_name).split("\n")
     nodes.remove('')
     return nodes
 
+def expand_nodes(node_list):
+    tmp_file = NamedTemporaryFile(dir='/tmp/')
+    with open(tmp_file.name, 'w') as fil:
```

Comment:
Either shorten file to be a single character `f` OR have it in its full form `file`. Otherwise `fil` is ambiguous to its meaning. It is acceptable to use a single character `f` as it is a single line block (unless python has a different standard). 

You maybe able to condense this down to a single line, I'm not sure what the line length is

Reply:
Ah yep, sorry, i'll change it to f (which is entirely with in standard). 
I was annoyed while writing it & file is a reserved word so I just deleted the last char.